### DISCUSSION TITLE
Feature Request: Ability to use Macros in Request Body

### DISCUSSION THREAD

**Discussion by TheJaysH** [grafana/grafana-json-datasource#257] [Ideas]
The api I'm using (Azure Billing API) requires date range to be sent in the json body of the request:

```json
{
  "type": "Usage",
  "timeframe": "Custom",
  "timePeriod": {
      "from": "2021-05-17T20:48:09.000Z",
      "to": "2021-05-17T20:48:09.000Z"
  },
  "dataset": {
    "granularity": "None",
    "aggregation": {
      "totalCost": {
        "name": "PreTaxCost",
        "function": "Sum"
      }
    },
    "grouping": [
      {
        "type": "Dimension",
        "name": "ResourceGroup"
      }
    ]
  }
}
```

It would be really nice to be able to use the date macro's in the json body like so:

```json
  "timePeriod": {
      "from": "$__isoFrom()",
      "to": "$__isoTo()"
  },
```

**Comment by layoaster** [comment]
Also interested in having these macros as variables for Grafana dashboards. 

My use case involves SQL queries sent as payloads so it would save from having to call a time format conversion function in the SQL server.

I could translate expressions (contained in the request body) such as: 

```sql
SQL *
FROM datasource
WHERE __time >= MILLIS_TO_TIMESTAMP(${__from}) AND __time < MILLIS_TO_TIMESTAMP(${__to})
```
into
```sql
SQL *
FROM datasource
WHERE __time >= '${__isoFrom}') AND __time < '${__isoTo}'
```

And also saving computation on the SQL server side.

**Comment by bakman2** [comment]
Same here. I am using surrealdb, which has build-in rest/websockets endpoints. 

Currently I have to do timerange conversions with a translator (node-RED) in between. 
Would be great if the macros could be used in the body directly, so I can get rid of the translator.   

Trying to decypher the code, is it not just a matter of calling `replace(body)` on the interpolate ?
